javascript/sort_ints/test.js

```diff
@@ -0,0 +1,40 @@
+const sorter = require('./solution');
+
+describe('The Integer Sorting Kata', () => {
+  test('should return empty array with null input', () => {
+    _runSort(null, []);
+  });
+
+  test('should return empty array with empty array input', () => {
+    _runSort([], [])
+  });
+
+  test('should return the array with single item array input', () => {
+    _runSort([1], [1]);
+  });
+
+  test('should return the array with a sorted, two item array input', () => {
+    _runSort([1,2], [1,2]);
+  });
+
+  test('should return a sorted array with an unsorted, two item array input', () => {
+    _runSort([2,1], [1,2]);
+  });
+
+  test('should return the array with a sorted, three item array input', () => {
+    _runSort([1,2,3], [1,2,3]);
```

Comment:
Here's a random question. Is there a reason why you wouldn't do it like this?
```suggestion
    _runSort([1,2,3], [1,2,3]);
    _runSort([1,3,2], [1,2,3]);
    _runSort([2,1,3], [1,2,3]);
    _runSort([2,3,1], [1,2,3]);
    _runSort([3,1,2], [1,2,3]);
    _runSort([3,2,1], [1,2,3]);
```

Reply:
Should move through one test case at a time. Each of those test cases can be solved by a different algorithm. Stepping through each of the, going from easiest to hardest (`[1,2,3]` is probably easiest, `[3,2,1]` is probably hardest) with drive the algorithm towards the final solution."

Reply:
OK, that makes sense.